packages/shared/scope-hoisting/src/generate.js

```diff
@@ -99,20 +99,20 @@ export function generate({
     ),
   );
 
-  let {code, rawMappings} = babelGenerate(ast, {
+  let generateResult = babelGenerate(ast, {
     sourceMaps: !!bundle.env.sourceMap,
     minified: bundle.env.minify,
     comments: true, // retain /*@__PURE__*/ comments for terser
   });
 
   let map = null;
-  if (bundle.env.sourceMap && rawMappings != null) {
+  if (bundle.env.sourceMap && generateResult.map != null) {
     map = new SourceMap(options.projectRoot);
-    map.addIndexedMappings(rawMappings);
+    map.addRawMappings(generateResult.map);
```

Comment:
This should be faster and more reliable then using `generateResult.rawMappings` as sending a single string from JS to CPP is faster than sending thousands of objects.

Parsing VLQs in C++ is also insanely fast so there's no noticeable performance impact in decoding these.

Reply:
The `generateResult.map` also no longer contains the sourceContents, so we need to figure out a way to get the content embedded in sourcemaps of the assets